### PR TITLE
4.0 - Overview, Beta and Upgrade guides

### PR DESCRIPTION
**Pull request by stephybun**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

### CODE REVIEW

**Review by manicminer — APPROVED** [review]
Just a minor observation on provider naming but otherwise LGTM! 🚀

File: website/docs/guides/4.0-upgrade-guide.html.markdown [hunk 1]
```diff
@@ -0,0 +1,1242 @@
...
+* This resource will need to be recreated upon upgrading so that the tag revision can be included in the resource ID.
+
+### `azurerm_api_management`
+
+* The deprecated block `policy` has been removed in favour of the `azurerm_api_management_policy` resource.
+
+### `azurerm_application_insights`
+
+* The property `daily_data_cap_in_gb` now defaults to `100`.
+* The property `"daily_data_cap_notifications_disabled` now defaults to `false`.
```
Comment (jackofallops) [inline website/docs/guides/4.0-upgrade-guide.html.markdown:633]:
```suggestion
* The property `daily_data_cap_notifications_disabled` now defaults to `false`.
```

File: website/docs/guides/4.0-upgrade-guide.html.markdown [hunk 2]
```diff
@@ -0,0 +1,1242 @@
...
+### `azurerm_linux_function_app`
+
+* The properties `site_config.health_check_path` and `site_config.health_check_eviction_time_in_min` must be set together.
+
+### `azurerm_linux_virtual_machine_scale_set`
+
+* The deprecated block `gallery_applications` has been removed in favour of the `gallery_application` block.
+* The deprecated block `terminate_notification` has been removed in favour of the `termination_notification` block.
+* The deprecated property `scale_in_policy` has been removed in favour of the `scale_in` block.
+* The property `extention_operations_enabled` now defaults to `true`.
```
Comment (jackofallops) [inline website/docs/guides/4.0-upgrade-guide.html.markdown:843]:
```suggestion
* The property `extension_operations_enabled` now defaults to `true`.
```

File: website/docs/guides/4.0-upgrade-guide.html.markdown [hunk 3]
```diff
@@ -0,0 +1,1242 @@
...
+
+* The property `version` now defaults to `1`.
+
+### `azurerm_nginx_deployment`
+
+* The deprecated block `configuration` has been removed in favour of the `azurerm_nginx_configuration` resource.
+
+### `azurerm_orchestrated_virtual_machine_scale_set`
+
+* The property `extention_operations_enabled` now defaults to `true`.
```
Comment (jackofallops) [inline website/docs/guides/4.0-upgrade-guide.html.markdown:944]:
```suggestion
* The property `extension_operations_enabled` now defaults to `true`.
```

File: website/docs/guides/4.0-upgrade-guide.html.markdown [hunk 4]
```diff
@@ -0,0 +1,1242 @@
...
+### `azurerm_windows_function_app`
+
+* The properties `site_config.health_check_path` and `site_config.health_check_eviction_time_in_min` must be set together.
+
+### `azurerm_windows_virtual_machine_scale_set`
+
+* The deprecated block `gallery_applications` has been removed in favour of the `gallery_application` block.
+* The deprecated block `terminate_notification` has been removed in favour of the `termination_notification` block.
+* The deprecated property `scale_in_policy` has been removed in favour of the `scale_in` block.
+* The property `extention_operations_enabled` now defaults to `true`.
```
Comment (jackofallops) [inline website/docs/guides/4.0-upgrade-guide.html.markdown:1123]:
```suggestion
* The property `extension_operations_enabled` now defaults to `true`.
```

**Review by jackofallops — APPROVED** [review]
Thanks @stephybun - just a couple typos and a couple suggestions, but otherwise LGTM and great work pulling all this together!

File: website/docs/guides/4.0-overview.html.markdown [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  }
+}
+
+# Configure the Microsoft Azure Provider
+provider "azurerm" {
+  features {}
+}
+```
+
+## What's available in Version 4.0 of the AzureRM Provider?
```
[the remark this answers is not in the document]
Comment (stephybun) [reply website/docs/guides/4.0-overview.html.markdown:44]: I'm on board with this!

**Review by jackofallops — APPROVED** [review]
Thanks @stephybun - one last pass and everything LGTM 🐫

**Pull request merged**